### DISCUSSION TITLE
Improve Stability and Crash Recovery in OpenToonz

### DISCUSSION THREAD

**Discussion by farhad1-max** [opentoonz/opentoonz#6366] [General]
### Is your feature request related to a problem? Please describe

I often encounter crashes while using OpenToonz, especially during workflows involving specific file types (e.g., importing video files or using multimedia codecs). While I appreciate the current functionality, these crashes disrupt my work and can result in lost progress. It would be great to have more robust stability and crash recovery features.

### Describe the solution you'd like

Auto-Save Enhancements:

Introduce a more frequent or customizable auto-save feature, allowing users to set time intervals for backups.
Save multiple backup versions to reduce the risk of losing data after a crash.
Crash Recovery Tool:

Implement a feature that automatically restores the most recent session/project after a crash.
Provide a clear crash recovery prompt upon reopening the software.
Improved Error Messaging:

Display detailed, user-friendly error messages when a crash occurs, identifying potential causes (e.g., problematic files or plugins).
Suggest troubleshooting steps based on the crash type.
Enhanced Codec Support:

Address compatibility issues with multimedia codecs (e.g., lvcod64.dll or MSVFW32.dll) to reduce crashes related to video and sound file imports.
Stability Improvements for Plugins:

Improve compatibility and error handling for third-party plugins or drivers, such as those from Logitech.
Additional Context:
Including a crash recovery system and more robust error handling will significantly enhance the user experience, especially for those working on complex projects. OpenToonz is already a powerful tool, and these improvements would make it more reliable for professionals and hobbyists alike.

**Comment by RodneyBaker** [comment]
Thanks for the feedback @farhad1-max 

Improved stability is a regular goal of development** but as we are now in the 2025 development cycle I've added the roadmap label to help us emphasize that review of stability issues moving forward.

Thanks also for identifying specific areas of concern to focus upon.

**In other words, stability isn't a feature request as much as it is a regular requirement of code review and development.  What isn't as routine are the elements that target specific stability issues and move issues of stability farther away from being commonly encountered by users.

**Comment by flurick** [comment]
For sure having work disappear is a no go for most. There are some alternatives in OT already that are maybe not used as much as they could. 
 
- [x] Autosave. There was something that made it break more things than it saved at times?
- [x] Backup files (.bak) Not sure when they are generated, not often enough to be useful?
- [x] SVN version manager. But needs a svn repository to connect to.
- [x] External project files backup. Needs to be setup - Windows "File Version", macOS "Time Machine ", Linux rsync etc

Some options:
- [ ] Save empty new projects so they can auto-save or which ever option works best
- [ ] Make and keep more .bak files around with auto-save
- [ ] Include svn server tool when installing
- [ ] "New" project portable file-format with history
